src/server/page.tsx

```diff
@@ -145,22 +164,33 @@ function page(
 ): Page {
     const item = fromCapi({ docParser, salt: imageSalt })(content);
     const shouldHideAds = content.fields?.shouldHideAdverts ?? false;
+    const hasTwitter = includesTweets(content);
     const clientScript = scriptName(item).fmap(getAssetLocation);
-    const html = (
-        <html lang="en" css={PageStyles}>
+    const emotionCache = createCache();
+    const { extractCritical } = createEmotionServer(emotionCache);
+    const { html: body, css, ids } = compose(extractCritical, renderToString)(
```

Comment:
What benefit does this give us?

Reply:
> Extracted Emotion into single style tag for hashing

By default Emotion scatters `<style>` tags across the DOM. To prevent the CSP from blocking these we'd need to somehow find all of these tags and hash their content. Achieving this would mean using one of a number of techniques, e.g. walking the React tree, applying string searches/regexes to the HTML string, or parsing the whole tree into a DOM and extracting them using the DOM APIs. I felt these all had greater complexity/performance downsides than using the built-in Emotion API to extract the CSS.